dfastmi/gui/dialog_view.py

```diff
@@ -1069,32 +1077,29 @@ def _update_plotting(self) -> None:
         ---------
         None
         """
-        if self._view_model.model.plotting != self._make_plots_edit.isChecked():
-            self._view_model.model.plotting = self._make_plots_edit.isChecked()
-            self._update_enabling_plotting_controls()
+        if self._view_model.make_plot != self._make_plots_edit.isChecked():
+            self._view_model.make_plot = self._make_plots_edit.isChecked()
 
-    def _update_enabling_plotting_controls(self):
-        self._save_plots.setEnabled(self._view_model.model.plotting)
-        self._save_plots_edit.setEnabled(self._view_model.model.plotting)
-        self._close_plots.setEnabled(self._view_model.model.plotting)
-        self._close_plots_edit.setEnabled(self._view_model.model.plotting)
+    def _update_enabled_of_make_plot_dependent_view_items(self, value: bool):
+        self._save_plots.setEnabled(value)
+        self._save_plots_edit.setEnabled(value)
+        self._close_plots.setEnabled(value)
+        self._close_plots_edit.setEnabled(value)
+
+    def _update_enabled_of_save_plot_dependent_view_items(self, value: bool):
+        self._figure_dir.setEnabled(value)
+        self._figure_dir_edit.setEnabled(value)
+        figure_dir_button = self._general_widget.findChild(
+            QPushButton, "figure_dir_edit_button"
+        )
+        figure_dir_button.setEnabled(value)
 
     def _update_save_plotting(self) -> None:
         """Update the plotting flags."""
 
-        save_plot_gui = (
-            self._save_plots_edit.isChecked() and self._view_model.model.plotting
-        )
-
-        if self._view_model.model.save_plots != save_plot_gui:
```

Comment:
this line should remain i think.

Reply:
reverted back with the discussed change.